user/syscalls.c

```diff
@@ -56,9 +100,22 @@ ssize_t write(int fd, const void *buf, size_t count) {
   return retval;
 }
 
-_off_t lseek(int __fildes, _off_t __offset, int __whence) {
-  errno = ENOSYS;
-  return (off_t)-1;
+_off_t lseek(int fd, _off_t offset, int whence) {
```

Comment:
Please rewrite this file to assembly. In most cases the code will be shorter and clearer:

```
#define DO_SYSCALL(num)         \
        li      $v0, num        \
        syscall                 \
        bgez    $v0, 1f         \
        move    $v1, $0         \
        sub     $v1, $0, $v0    \
        addi    $v0, $0, -1     \
1:      jr      $ra             \
        sw      $v1,_errno($gp) \

LEAF(_lseek)
      DO_SYSCALL(SYS_LSEEK)
END(_lseek)
```